### PR TITLE
feat(tbgenkit): Add tbgenkit - a new package for Genkit Go

### PR DESCRIPTION
**Pull request by dishaprakash**
feat(tbgenkit): Add tbgenkit - a new package for Genkit Go

This PR adds a new package in the mcp-toolbox-sdk-go module.
This package contains a function `ToGenkitTool` which converts a ToolboxTool into a Genkit compatible tool

The necessary tests are also added as a part of this PR

### CODE REVIEW

**Comment by dishaprakash** [comment]
I have not added detailed tests for edge cases as those are covered under core and the genkit module.

File: tbgenkit/tbgenkit.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	// Retrieve the JSON schema bytes from the custom tool.
+	jsonBytes, err := tool.InputSchema()
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal the JSON schema bytes into a jsonschema.Schema object.
+	var schema *jsonschema.Schema
+	if err := json.Unmarshal(jsonBytes, &schema); err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (dishaprakash) [reply tbgenkit/tbgenkit.go]: Wrapped with a comprehensive error

File: tbgenkit/tbgenkit_test.go [hunk 2]
```diff
@@ -0,0 +1,464 @@
+//go:build tbgenkit
```
[the remark this answers is not in the document]
Comment (dishaprakash) [reply tbgenkit/tbgenkit_test.go]: Thanks for catching this! I've changed the workflow to run tests based on packages now
